src/shared/Microsoft.Git.CredentialManager/Settings.cs

```diff
@@ -55,20 +90,109 @@ public Settings(IEnvironmentVariables environmentVariables, IGit git)
 
         public bool IsSecretTracingEnabled => _environment.GetBooleanyOrDefault(KnownEnvars.GcmTraceSecrets, false);
 
+        public bool IsMsalTracingEnabled => _environment.GetBooleanyOrDefault(Constants.EnvironmentVariables.GcmTraceMsAuth, false);
+
+        public bool IsCertificateVerificationEnabled =>
+            TryGetSetting(null, KnownGitCfg.Http.SectionName, KnownGitCfg.Http.SslVerify, out string value) && !value.ToBooleanyOrDefault(true);
+
+        public Uri ProxyConfiguration
+        {
+            get
+            {
+                bool TryGetUriSetting(string envarName, string section, string property, out Uri uri)
+                {
+                    IEnumerable<string> allValues = GetSettingValues(envarName, section, property);
+
+                    foreach (var value in allValues)
+                    {
+                        if (Uri.TryCreate(value, UriKind.Absolute, out uri))
+                        {
+                            return true;
+                        }
+                    }
+
+                    uri = null;
+                    return false;
+                }
+
+                /*
+                 * There are several different ways we support the configuration of a proxy.
+                 *
+                 * In order of preference:
+                 *
+                 *   1. GCM proxy Git configuration
+                 *        credential.httpsProxy
+                 *        credential.httpProxy
+                 *
+                 *   2. Standard Git configuration
+                 *        https.proxy
+                 *        http.proxy
+                 *
+                 *   3. cURL environment variables
+                 *        HTTPS_PROXY
+                 *        HTTP_PROXY
+                 *        ALL_PROXY
+                 *
+                 *   4. Legacy GCM proxy environment variable
+                 *        GCM_HTTP_PROXY
+                 *
```

Comment:
Ideally how many ways to set the proxy would we want to support/need?

Reply:
Good question.. cc: @jrbriggs

I've gone for maximal compatibility here currently.